src/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,131 @@
+"""
+    GibbsConditional(sym, conditional)
+
+A "pseudo-sampler" to manually provide analytical Gibbs conditionals to `Gibbs`.
+`GibbsConditional(:x, cond)` will sample the variable `x` according to the conditional `cond`, which
+must therefore be a function from a `NamedTuple` of the conditioned variables to a `Distribution`.
+
+Example:
+
+```julia
+α₀ = 2.0
+θ₀ = inv(3.0)
+
+x = [1.5, 2.0]
+
+function gdemo_statistics(x)
+    # The conditionals and posterior can be formulated in terms of the following statistics:
+    N = length(x) # number of samples
+    x̄ = mean(x) # sample mean
+    s² = var(x; mean=x̄, corrected=false) # sample variance
+    return N, x̄, s²
+end
+
+function gdemo_cond_m(c)
+    # c = (λ = ...,)
+    N, x̄, s² = gdemo_statistics(x)
+    mₙ = N * x̄ / (N + 1)
+    λₙ = c.λ * (N + 1)
+    σₙ = √(1 / λₙ)
+    return Normal(mₙ, σₙ)
+end
+
+function gdemo_cond_λ(c)
+    # c = (m = ...,)
+    N, x̄, s² = gdemo_statistics(x)
+    αₙ = α₀ + (N - 1) / 2
+    βₙ = (s² * N / 2 + c.m^2 / 2 + inv(θ₀))
+    return Gamma(αₙ, inv(βₙ))
+end
+
+@model gdemo(x) = begin
+    λ ~ Gamma(α₀, θ₀)
+    m ~ Normal(0, √(1 / λ))
+    x .~ Normal(m, √(1 / λ))
+end
+
+m = gdemo(x)
+
+sample(m, Gibbs(GibbsConditional(:λ, gdemo_cond_λ), GibbsConditional(:m, gdemo_cond_m)), 10)
+```
+"""
```

Comment:
Very good documentation!

Do you think it would make sense to write the code example as a doctest, as suggested in the [Julia docs](https://docs.julialang.org/en/latest/manual/documentation/)? BTW apparently it's also suggested to put code examples in a `# Examples` section, so maybe that could be changed.

Reply:
Aren't doctests a bad idea for programs with random output?  Or are they working with a fixed seed automatically?

Reply:
Let's keep it as it is and have a discussion about using doctest in the code base on slack.

Reply:
> Aren't doctests a bad idea for programs with random output? Or are they working with a fixed seed automatically?

It's not reproducible automatically but of course one can explicitly specify a RNG. From the docs:

> Calling rand and other RNG-related functions should be avoided in doctests since they will not produce consistent outputs during different Julia sessions. If you would like to show some random number generation related functionality, one option is to explicitly construct and seed your own MersenneTwister (or other pseudorandom number generator) and pass it to the functions you are doctesting.